File: Semester 4/Advanced Programming/ClientServerApplication/ApplicationRepository/src/main/java/app/repository/UserDbRepo.java
package app.repository;

import app.model.User;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.List;
import java.util.Properties;

public class UserDbRepo implements IUserRepo {
    private DbUtils dbutils;
    private Logger logger = LogManager.getLogger();
    public UserDbRepo(Properties props) {
        logger.info("Constructing UserDbRepo with properties : {}",props);
        dbutils=new DbUtils(props);
    }
    @Override
    public void save(User entity) {

    }

    @Override
    public void delete(String s) {

    }

    @Override
    public void update(User entity) {

    }

    @Override
    public User findOne(String id){
        logger.traceEntry("Finding User with key : {}",id);
        Connection con = dbutils.getConnection();
        try(PreparedStatement stm = con.prepareStatement("select * from Users where id =?")){
            stm.setString(1,id);
            try(ResultSet result = stm.executeQuery()){
                if(result.next()){
                    String id2 = result.getString("id");
                    String pass = result.getString("password");
                    return new User(id2,pass);
                }
            }
        }catch (SQLException ex){
            logger.error(ex);
        }
        logger.traceExit();
        return null;
    }

    @Override
    public List<User> findAll() {
        return null;
    }
}

File: Semester 4/Advanced Programming/RpcProtocolC#/Repository/UserDbRepo.cs
using log4net;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Domain;
namespace Repositories
{
    public class UserDbRepo : IUserRepo
    {
        private static readonly ILog log = LogManager.GetLogger("UserDbRepo");
        public UserDbRepo()
        {
            log.Info("Creating UserDbRepo");
        }

        public void Delete(string id)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<User> FindAll()
        {
            throw new NotImplementedException();
        }

        public User FindOne(string id)
        {
            Console.WriteLine("Enter function findOne with key " + id);
            var connection = DBUtils.getConnection();
            using (var cmd = connection.CreateCommand())
            {
                cmd.CommandText = "Select * from Users where id=@id";
                var param = cmd.CreateParameter();
                param.ParameterName = "@id";
                param.Value = id;
                cmd.Parameters.Add(param);
                using (var dataReader = cmd.ExecuteReader())
                {
                    if (dataReader.Read())
                    {
                        string username = dataReader.GetString(0);
                        string password = dataReader.GetString(1);
                        connection.Close();
                        return new User(username, password);
                    }
                }
            }
            return null;
        }

        public void Save(User entity)
        {
            throw new NotImplementedException();
        }

        public void Update(User entity)
        {
            throw new NotImplementedException();
        }
    }
}
